manifests/unit_file.pp

```diff
@@ -90,6 +90,14 @@
 
   assert_type(Systemd::Unit, $name)
 
+  if $hasrestart {
+    warning("systemd::unit_file - ${title}: hasrestart is deprecated and will be removed in Version 4 of the module")
```

Comment:
Does it make sense to use `deprecation()` from stdlib for this?

Reply:
I wasn't aware of that. I will take a look at what it does and use it

Reply:
I implemented `deprecation()`. Depending on the puppetserver configuration nothing will be logged:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/ea43f2ecd217ff85fe6797b177c7df7718266a94/lib/puppet/functions/deprecation.rb#L29-L38

should we keep the `warning()` because of that, or remove it?

Reply:
I'd say remove it. @alexjfisher usually has some good ideas about this though.

Reply:
removed it for now..